pkg/controller/networkpolicy/status_controller.go

```diff
@@ -336,6 +336,8 @@ func (c *StatusController) syncHandler(key string) error {
 	phase := crdv1alpha1.NetworkPolicyRealizing
 	if currentNodes == desiredNodes {
 		phase = crdv1alpha1.NetworkPolicyRealized
+	} else if currentNodes+len(failedNodes) == desiredNodes {
+		phase = crdv1alpha1.NetworkPolicyFailed
 	}
```

Comment:
Why this calculation is needed? we may just check failedNodes only like below:
```
if len(failedNodes) > 0 {
  phase = crdv1alpha1.NetworkPolicyFailed
}
```

Reply:
Question - in K8s Node case, do we have permanent realization failure - that is agent already gave up retrying realizing?

I agree "failed" and "realized" seems not match, but on the other hand people also wonder why we just have "realized" which seems a condition too (rather a phase), but no phase to express failure.

Reply:
For Pod case:
- If the Pod's restart policy is never
  * If any of containers is still running, the phase is Running
  * If none of containers is running, and there is any failure, the phase is Failed
  * If none of contaienrs is running, and there is no failure, the phase is Succeeded
- If the Pod's restart policy is always
  * The phase is running regardless whethere there is running Pod or failed Pod

Its principle seems: if the system keeps trying to reach desired state, the phase will always be "XXXing"; if the system is asked to not retry, the phase will be "XXXed".

Reply:
In our case, does agent always keep realizing?

Reply:
Antrea Agent always try to realize the ANP rules. Nephe may set "RealizationFailure: false" when the ANP rules are failed to realize on cloud.